src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -1306,6 +1310,32 @@ internal class PipelineStopper
         /// Object used for synchronization.
         /// </summary>
         private object _syncRoot = new object();
+
+        internal void AwaitPipelineCriticalSection()
+        {
+            _criticalSectionSemaphore.Wait();
+            _criticalSectionSemaphore.Release();
+        }
```

Comment:
Yeah. Hm. This is utilized by `StopProcessing()` to ensure pipeline stops don't cancel an in-process `cleanup{}`.

I'm not sure if we'd want to have a StopProcessing() call itself be treated as a critical region? My instinct is to say we _don't_, just because currently there's nothing stopping multiple StopProcessing() calls in quick succession before one completes. This retains as much behaviour from the current state of things as possible without letting StopProcessing() cancel a pipeline where we don't want it to.